fmlcore/src/main/java/net/minecraftforge/fml/config/ModConfig.java

```diff
@@ -86,7 +87,14 @@ public Path getFullPath() {
     }
 
     public void acceptSyncedConfig(byte[] bytes) {
-        setConfigData(TomlFormat.instance().createParser().parse(new ByteArrayInputStream(bytes)));
+        if (bytes != null)
+        {
+            setConfigData(TomlFormat.instance().createParser().parse(new ByteArrayInputStream(bytes)));
+        }
+        else
+        {
+            setConfigData(null);
```

Comment:
The reason why I did it this way is that this brings it back to the default state that should exist on the main menu yes we don't have an unloaded event but I am not sure that is the right word for that either.

The thing this mainly does is remove the server config.
Obviously, there can be no server config if there is no world, if somebody would be accessing the server configs on the main menu for example it would already crash today.

So yes you could consider adding one, but it is in my eyes also not a breaking change. It is just a clarification, they are not supposed to exist anyway in the first place.